conans/client/graph/graph_manager.py

```diff
@@ -100,6 +100,7 @@ def load_consumer_conanfile(self, conanfile_path, info_folder,
             if test:
                 conanfile.display_name = "%s (test package)" % str(test)
                 conanfile.output.scope = conanfile.display_name
+                conanfile.testing_reference = str(test)
```

Comment:
Why ``str()`` and not the RecipeReference object? Why not ``repr()`` so we don't lose potential information like the recipe revision?

Reply:
Yes, I was expecting this comment to talk about it:

 If you do a `self.requires(self.testing_reference)` it fails because it expects an `str`. So we would need to enable a `ConanFileReference` as a valid argument. 

Also, I introduced the `str` here because I'm not sure we have documented the `ConanFileReference` (maybe in the `dependencies`?) and it is probably that the users are going to compare the reference so we have to know that `ConanFileReference.loads("foo/1.0") != "foo/1.0"`. So I wanted to talk about this before taking the decission.